pkg/ttnpb/fieldmask_utils.go

```diff
@@ -72,6 +72,9 @@ nextRequested:
 
 // HasAnyField returns whether the given requested paths contain any of the given fields.
 // The requested fields (i.e. `a.b`) may be of a higher level than the search path (i.e. `a.b.c`).
+//
+// Note that this function may have unexpected results when non bottom search fields are used,
+// as HasAnyField([]string{"a.b"}, "a") is false.
```

Comment:
This is because `a.b` doesn't imply the entire `a`. So if all possibilities are `[a, a.b, a.c]`, and we have `[a.b]`, then requesting `[a]` should be false because if it would be true, then `a.c` can be expected.